### DISCUSSION TITLE
Hardware recommendations

### DISCUSSION THREAD

**Discussion by templeowls21** [blakeblackshear/frigate#14693] [Ask A Question]
I currently run Frigate via the HA add-on on my HA server. I'd like to seperate it to its own device.

I have 10 IP cameras and want to record 24/7 with a 5 day retention. Plus detection and 30 day event retention period.

Based on the Frigate hardware recs, I was thinking a Beelink EQ12 plus an additional 2TB SSD and a Coral USB.

Is this device sufficient for my number of cameras / retention? Would a NUC be needed? Is a SSD okay or should it be a HDD? Thanks!!

**Comment by NickM-27** [comment]
> Based on the Frigate hardware recs, I was thinking a Beelink EQ12 plus an additional 2TB SSD and a Coral USB.

that is good, although openvino would work great and you likely wouldn't need a coral

**Reply by templeowls21** [reply]
Thanks! So no need for a NAS, right? An internal or external SSD would work fine?

**Reply by NickM-27** [reply]
yes

**Reply by templeowls21** [reply]
Gotcha. Would a HDD be better you think? I've read that SSDs struggle with constant rewrites but idk if that is BS

**Reply by NickM-27** [reply]
I don't believe that is really true, at least not any more. I have had 8 cameras writing to SSDs for multiple years without any issue

**Reply by leccelecce** [reply]
It's not BS. It really depends on the SSD (and things which determine write MB/s e.g. # of cameras, video resolution, H264 vs H265 etc). If you buy the cheapest 2TB drive you can find from a no-name Chinese brand, it will clearly be more of a problem than Samsung, Intel drives etc. 

You need to look at the endurance scores for the drive which can be expressed as DWPD or TBW/PBW and see how quickly you will reach them. If you go beyond that number, you are gambling your video, particularly if you have 1 drive and no backup. Consumer SSDs are simply not designed to have the whole contents of the drive written every day, because almost nobody does that. Temperature can be an issue as well. You can pick up datacentre SSDs (Intel S4500 1.92tb for example) that are built for constant write-heavy workloads and then you don't have to worry about it.

Whatever you do go for, the important thing is to monitor the drive health % via SMART. I have seen cheap NVMe drives drop from 100% to 80% in <6 months, and start throwing various NVMe write errors in the kernel log at that point. Any good manufacturer should give you some indication of how the drive is holding up.

If you're looking for a cheap option, just be willing to monitor drive health month to month and potentially replace drives in a couple of years and you should be fine. But I wouldn't view it as "set and forget". I run 2x 4TB WD Purple drives in RAID1, different manufacturing years; more noise, heat and power consumption but I have zero tolerance for data loss. Snapshots and the Frigate DB are also rsync'd off site (video is too much bandwidth).

**Reply by templeowls21** [reply]
Thanks! Do you mind me asking which NAS you have? I already has a DS223j with 2 WD Reds but this is the lowest entry in Synology's lineup & i don't feel its adequate for video recording storage & constant rewrites for 10 cameras

It's embarrassing to admit this is why, but I'm trying to avoid a NAS because I'm maxed out of space in my networking closet. Ultimately, I can figure it out if needed, but if I can get a standalone SSD/HDD which does the job, no matter the cost, I'd love that

**Reply by leccelecce** [reply]
I have a Terramaster U4-423 which is 4 drives in 1U. It would be overkill in terms of horsepower for Frigate alone but the other two drives +NVMe flash are for VM storage. 2x2.5Gbe and quad core CPU can get maxed out when running backups, snapshots etc. Most of the day it's barely doing anything other than ~150mbps writing video for Frigate.

I would have thought your DS223j would cope with Frigate's requirements, but if you're using it for other purposes as well you'll probably see a bit of latency. You should note though that older WD Red drives (SMR) are specifically recommended against for 24x7 workloads these days after a reported large number of failures (new Red Pro or Red Plus are fine).

**Reply by templeowls21** [reply]
Thank you! I use the DS223j currently for just basic, minimal file storage. So I tested it out with Frigate (HA add-on and DS223j is mounted in HA) and the CPU usage goes extremely high (80%). I posted it in the Synology forums and, the technical stuff they said was a bit over my head, but they basically said 10 cameras 24/7 are too much for it to handle. It's an entry level unit they said, but maybe its moreso the WD Reds or using it mounted in HA

**Reply by leccelecce** [reply]
https://www.reddit.com/r/synology/comments/1g46nwn/very_high_cpu_usage_for_camera_recordings/
So in this thread, there's quite a lot of misinformation, based on seemingly some confusion in the Synology forums on how Frigate works. You have people there thinking the NAS is connecting to your cameras (it isn't), equally some people are talking about running Frigate itself directly on the NAS in Docker (you're not).
I would be wary that your WD Red's are possibly the limitation rather than the NAS so if you upgrade it...make sure you can return it!

**Reply by templeowls21** [reply]
got it! Thanks! I'll pick up a WD Purple today and try that out before revamping anything else.

So you're saying the high CPU usage of the NAS could be due to inefficiencies of writing 24/7 on the WD Reds?

**Reply by leccelecce** [reply]
With WD Purple check the cache size is 256mb. The initial models e.g. WD40PURZ had 64mb, they released a newer WD43PURZ with 256mb. 

I don't think the change in disk will reduce your NAS CPU all that much. It should reduce latency when seeking as there is more cache to take writes. It's also smarter about how it writes. Finally it's a CMR drive and if your Red is a SMR model they have known issues with write performance (and reliability).

I can't guarantee any improvements without knowing the exact existing hardware, configuration and diagnostic outputs, but you won't lose anything by having an NVR class drive.

**Reply by templeowls21** [reply]
Gotcha. I'll at least try it out. The option near me is indeed 256; thanks for the heads up on that.

Let's say I still go down the route of upgrading everything...what would you ultimately suggest as the best storage option if running frigate on docker on the Beelink w/ Coral?
- 10 IP cameras 
- 24/7 recording at 1080p, 20fps with 5 day retention
- Event recording of detection with 30 day retention of substream (704x480, 10fps)

If you need more info, let me know. I really just want a set and forget (as much as possible) setup. Thanks for all the help!
